### PR TITLE
Docs: Refactor data sources content

### PR DESCRIPTION
**Pull request by gguillotte-grafana**
**What is this feature?**

Reorganizes the built-in data sources documentation to align in structure and verbiage.

**Why do we need this feature?**

Each built-in data source's documentation is structured differently, sometimes inconsistently so. There are opportunities for automatic content reuse across these docs (cc @jdbaldry), and each data source should provide the same context and format for configuration and usage.

Built-in data source content arbitrarily combines administrative content like adding and configuring data sources, which requires administrative user roles/privileges, with usage content like querying and annotating, which typically does not. This results in excessively long and difficult to navigate documents with no clear audience. This PR breaks those topics into separate documents, and does so consistently across most data sources.

(Conversely, some configuration content is arbitrarily separated from the rest of it into a separate document. This PR also merges those documents back.)

The files themselves are Markdown files inconsistently structured, or not, into Hugo bundles. This consistently moves all data source docs into bundles, which makes future expansion of these topics more consistent.

Some documentation in the /datasources/ directory is redundant with content now in /administration/. This PR removes that redundant content from /datasources/ and more consistently links to it in /administration/.

**Who is this feature for?**

- Grafana administrators, to provide more focused organization of content around built-in data sources.
- Grafana users, to present a clearer workflow for interacting with data sources.
- Our website team, to restructure the files into bundles that are more consistent with the rest of the docs' organization.

**Which issue(s) does this PR fix?**:

N/A

**Special notes for your reviewer**:

This is a large PR that revises and moves several files in separate commits. It might be easier to review this PR by following individual commits, which separates the revision and moving operations for more informative diffs and also separates revisions to related content outside of the data source docs.

### CODE REVIEW

File: docs/sources/datasources/_index.md [hunk 1]
```diff
@@ -2,51 +2,75 @@
...
+  - /docs/grafana/latest/data-sources/
 title: Data sources
 weight: 60
 ---
 
 # Data sources
 
-Grafana supports many different storage backends for your time series data (data source). Refer to [Add a data source]({{< relref "../administration/data-source-management/#add-a-data-source/" >}}) for instructions on how to add a data source to Grafana. Only users with the organization admin role can add data sources.
+Grafana supports many different backends for your time series data, and lets you create your own interfaces to backends.
+Grafana refers to these backends as **data sources**.
```
Comment (torkelo) [inline docs/sources/datasources/_index.md:14]:
```suggestion
Grafana can query and integrate with many different types of databases. This is done by adding a **data source** of the type you want to query or integrate with. 
```

File: docs/sources/datasources/_index.md [hunk 2]
```diff
@@ -2,51 +2,75 @@
...
 
-## Querying
+You can use this flexibility to build [panels]({{< relref "../panels-visualizations/" >}}) that collect and visualize data of many types, and from many sources, in a single [dashboard]({{< relref "../dashboards/" >}}).
+Each panel uses one specific data source, which can belong to a particular [organization]({{< relref "../administration/organization-management/" >}}).
```
Comment (torkelo) [inline docs/sources/datasources/_index.md:16]:
```suggestion
When you have created and configured a data source you are ready to start exploring and visualizing data, either in Explore or in a new Dashboard. A dashboard is composed of [panels]({{< relref "../panels-visualizations/" >}}), each panel contains a set of queries to one or more data sources. 

You can also create new a alert rule from a data source query and have Grafana continuously evaluate it and notify you when things change.

File: docs/sources/datasources/elasticsearch/_index.md [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+
+For instructions on how to add a data source to Grafana, refer to the [administration documentation]({{< relref "/administration/data-source-management/" >}}).
+Only users with the organization administrator role can add data sources.
+Administrators can also [configure the data source via YAML]({{< relref "#provision-the-data-source" >}}) with Grafana's provisioning system.
+
+> **Note:** You can use [Grafana Cloud](/products/cloud/features/#cloud-logs) to avoid the overhead of installing, maintaining, and scaling your observability stack.
+> The free forever plan includes Grafana, 10K Prometheus series, 50 GB logs, and more.
+> [Create a free account to get started](https://grafana.com/auth/sign-up/create-user?pg=docs-grafana-install&plcmt=in-text).
+
+Once you've added the Elasticsearch data source, you can [configure it]({{< relref "#configure-the-data-source" >}}) so that your Grafana instance's users can create queries in its [query editor]{{< relref "./query-editor/" >}} when they [build dashboards]({{< relref "../../dashboards/build-dashboards/" >}}) and use [Explore]({{< relref "../../explore/" >}}).
```
Comment (torkelo) [inline docs/sources/datasources/elasticsearch/_index.md:32]: Something wrong with the query editor link 
image

Comment (torkelo) [reply docs/sources/datasources/elasticsearch/_index.md:32]: same issue looks to be in all data source docs

File: docs/sources/panels-visualizations/_index.md [hunk 4]
```diff
@@ -16,11 +16,18 @@ keywords:
 
 # Panels and visualizations
 
-The _panel_ is the basic visualization building block in Grafana. Each panel has a query editor specific to the data source selected in the panel. The query editor allows you to extract the perfect visualization to display on the panel.
+The _panel_ is the basic visualization building block in Grafana.
+Each panel has a query editor specific to the data source selected in the panel.
+The query editor allows you to extract the perfect visualization to display on the panel.
```
Comment (torkelo) [inline docs/sources/panels-visualizations/_index.md:21]:
```suggestion
The query editor allows you to build a query to get the data you want to visualize. 
```

File: docs/sources/panels-visualizations/_index.md [hunk 5]
```diff
@@ -16,11 +16,18 @@ keywords:
...
 
-There are a wide variety of styling and formatting options for each panel. Panels can be dragged and dropped and rearranged on the dashboard. They can also be resized.
+There are a wide variety of styling and formatting options for each panel.
+Panels can be dragged, dropped, and resized to rearrange them on the dashboard.
 
-Before you begin, ensure that you have configured a data source. For more information about data sources, refer to [Data Sources]({{< relref "../administration/data-source-management/" >}}).
+Before you add a panel, ensure that you have configured a data source.
+While you can use the [special built-in "Grafana" data source]({{< relref "../datasources/#special-data-sources" >}}) to generate random data for testing, you can accomplish more with panels once you add a live data source.
```
Comment (torkelo) [inline docs/sources/panels-visualizations/_index.md:27]:
```suggestion
```

Not sure we need to mention the built in test data source and random data testing queries here

File: docs/sources/panels-visualizations/query-transform-data/_index.md [hunk 6]
```diff
@@ -19,158 +19,177 @@ weight: 200
 
 # Query and transform data
 
-Data source queries return data that can then be transformed via transformations and then visualized by different types of visualizations. The query language and query builder UI depends on the data source type. Grafana supports many different types of data sources.
+Grafana supports many types of [data sources]({{< relref "../../datasources/" >}}).
+Data source **queries** return data that Grafana can **transform** and visualize.
```
Comment (torkelo) [inline docs/sources/panels-visualizations/query-transform-data/_index.md:23]:
```suggestion
Data source **query** return data that Grafana can **transform** and visualize.
```

**Review by torkelo — COMMENTED** [review]
Looks great! left a few comments (and pushed a small fix to get it to compile). 

There are a lot of relref (links) that needs to be updated

**Review by torkelo — APPROVED** [review]
This looks great! huge improvements! :heart:

**Review by osg-grafana — APPROVED** [review]

**Pull request merged**